### PR TITLE
[TS] recommend option object args in docs

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

intentionally matching up with the go docs, teaching the simple form first, then mentioning that we recommend the struct form https://docs.temporal.io/docs/go/how-to-develop-a-workflow-definition-in-go

### CODE REVIEW

File: docs/typescript/workflows.md [hunk 1]
```diff
@@ -37,6 +37,14 @@ Workflow Definitions are "just functions", which can store state, and orchestrat
 <!--SNIPEND-->
 
 The snippet above uses `proxyActivities` to create functions that, when called, schedule a `greet` Activity in the system to say "Hello World".
+Workflow functions can support one or more arguments, but we encourage you to use single parameter option objects, as that helps with backward compatibility:
```
Comment (lorensr) [inline docs/typescript/workflows.md:40]:
```suggestion

A Workflow function can have multiple parameters, but we encourage you to use a single object parameter, as that helps with backward compatibility:
```

File: docs/typescript/workflows.md [hunk 2]
```diff
@@ -37,6 +37,14 @@ Workflow Definitions are "just functions", which can store state, and orchestrat
...
+
+```ts
+export async function example({ name: string }): Promise<{ name: string }> {
```
Comment (lorensr) [inline docs/typescript/workflows.md:43]:
```suggestion
export async function example({ name: string }): Promise<{ greeting: string }> {
```

File: docs/typescript/workflows.md [hunk 3]
```diff
@@ -38,6 +38,19 @@ Workflow Definitions are "just functions", which can store state, and orchestrat
...
+
+```ts
+export async function example({
+  name,
+}: {
+  name: string;
+}): Promise<{ greeting: string }> {
+  const greeting = await greet(name);
+  return { greeting };
+}
```
Comment (sw-yx) [inline docs/typescript/workflows.md:51]: maybe more readable?

```suggestion
type argsType = {
  name: string;
}

export async function example(args: argsType): Promise<{ greeting: string }> {
  const greeting = await greet(args.name);
  return { greeting };
}
```

Comment (lorensr) [reply docs/typescript/workflows.md:51]:
```suggestion
type ExampleArgs = {
  name: string;
}

export async function example(args: ExampleArgs): Promise<{ greeting: string }> {
  const greeting = await greet(args.name);
  return { greeting };
}
```

**Review by flossypurse — APPROVED** [review]

**Pull request merged**